src/pages/NewChairConsole.js

```diff
@@ -151,6 +190,30 @@ const NewChairConsole = () => {
     }
   };
 
+  const createElection = (darId, index) => {
+    if (darId !== null) {
+      let copy;
+      Election.createDARElection(darId)
+        .then(
```

Comment:
I just realized from our Object.assign conversation that you don't have to worry about null objects when dealing with a successful create because you should be getting the data back from the response. The issue however is that the Election.createDARElection doesn't work the way to expect since it's part of legacy code. The current function returns a raw response that you have to parse with .json(). However .json() in of itself is asynchronous, so you'll have to do some promise chaining

```
Election.createDARElection(darId)
.then(electionResponse => electionResponse.json())
.then(newElection => {
  //earlier code block//
  copy[index].election = newElection
}
```

The newElection data should be updated with the expected status and finalAccessVote values already, and it's already a different object so you can just assign it as the new election.

NOTE: @rushtong I'm not a fan of this approach, but the current implementation of createDARElection leaves us with little choice (await/async approach will have similar issues). createDARElection is used in AdminManageAccess, so I don't want Rae to deal with the function refactor in this PR (it can be handled in a separate PR). Is this reasonable?

Reply:
I see what you're saying, that makes sense to me!

Reply:
@JVThomas Good catch. The admin page will eventually be updated. We should create a new createElection method that works the way we want and deprecate the current function. Having the new function return the result of `await rawData.json()` (assuming a successful response) is the better way to go.